Core/clim-core/frames.lisp

```diff
@@ -244,6 +244,10 @@ documentation produced by presentations.")
       (setf (slot-value frame 'layouts) nil))
     (setf (%frame-manager frame) fm)))
 
+(defmethod (setf frame-pretty-name) :after (new-value frame)
+  (when-let ((frame-manager (frame-manager frame)))
+    (clime:note-frame-pretty-name-changed frame-manager frame new-value)))
```

Comment:
as per some convention I don't remember where I saw it (probably in CLIM II spec) note-* methods are reserved for api consumers (i.e application programmers), and clim should do its things inside the methods which perform the actual thing. In this spirit we shout set sheet's pretty name here, not in note-frame-pretty-name-changed.